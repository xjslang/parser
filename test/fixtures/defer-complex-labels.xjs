function testComplexDeferLabels() {
  console.log('testing complex defer vs labels')
  
  // Multiple defer statements
  defer console.log('cleanup 1')
  defer console.log('cleanup 2')
  
  // Labels with different spacing
  defer: {
    console.log('label with no space')
  }
  
  defer : {
    console.log('label with space before colon')
  }
  
  defer   :   {
    console.log('label with multiple spaces')
  }

  defer /* comment */ :
    /* comment */
    // comment
    // comment
    {
    console.log('label with comments')
  }
  
  // Nested labels
  outer: {
    defer: {
      console.log('nested defer label')
    }
    // This should still work as defer statement
    defer console.log('nested defer statement')
  }
  
  console.log('done')
}

testComplexDeferLabels()
