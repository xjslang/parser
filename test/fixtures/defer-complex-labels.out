testing complex defer vs labels
label with no space
label with space before colon
label with multiple spaces
label with comments
nested defer label
done
nested defer statement
cleanup 2
cleanup 1
